src/app/core/authentication/authentication.service.ts

```diff
@@ -35,6 +38,7 @@ export class AuthService {
             this.router.navigate(["/"], { queryParams: { checkPricing: true } });
 
             this.pricingTierService.calculateWalletAmount();
+            this.cartService.applyDiscount();
```

Comment:
I called the cart service here instead of in the pricing tier service since the CS already has the PTS injected, so injecting the CS into the PTS as well results in a circular dependency.